### ISSUE TITLE
Lateral column aliasing is not supported

### ISSUE THREAD

**Issue by damirbk** [sqlfluff/sqlfluff#5255]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

We have dbt models that use [lateral column aliasing](https://aws.amazon.com/about-aws/whats-new/2018/08/amazon-redshift-announces-support-for-lateral-column-alias-reference/). It is supported by at least Redshift, [Databricks](https://www.databricks.com/blog/introducing-support-lateral-column-alias)/Spark and ClickHouse. However, SQLFluff doesn't seem to understand it and treats it as a parsing error.

### Expected Behaviour

`sqlfluff lint` doesn't report any errors.

### Observed Behaviour

SQLFluff reports Parsing Error.

### How to reproduce

test.sql:
```sql
select
    (1 as one) + 1 as col_1,
    one + 3 as col_2,
    one + 5 as col_3,
    one + 7 as col_4

```
With the CLI, run:
`sqlfluff lint test.sql --dialect=redshift`
`sqlfluff lint test.sql --dialect=databricks`
`sqlfluff lint test.sql --dialect=clickhouse`

At the very least in ClickHouse, this is valid SQL.

SQLFluff fails with the following message:
```
== [test.sql] FAIL                                                                                                                                                      
L:   2 | P:   7 |  PRS | Line 2, Position 7: Found unparsable section: ' as one'
WARNING: Parsing errors found and dialect is set to 'clickhouse'. Have you configured your dialect correctly?
```

### Dialect

Redshift, Databricks, ClickHouse

### Version

SQLFluff version 2.3.2
Python version 3.10.10

### Configuration

```
[sqlfluff]
dialect = clickhouse
```

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
